### ISSUE TITLE
Specifying generated enum value types

### ISSUE THREAD

**Issue by maxmalmgren** [jburzynski/TypeGen#40]
In my project we use enums serialized as strings. It would be great if it was possible to specify string value generation, to support the following generated TypeScript:

```
export enum EnumExample {
    ExampleValue1 = "ExampleValue1",
    ExampleValue2 = "ExampleValue2"
}
```
Such a feature should probably include the possibility to support different casings between name and value.

I didn't find anything in `GeneratorOptions` which hints to this.

Thanks for a neat piece of software!

**Comment by jburzynski** [comment]
You're welcome!

This would be useful. I've added it as a feature for the next release, which will probably be out somewhere towards the end of December.

**Comment by jburzynski** [comment]
I added enum string initializers in 2.1.0. You can switch them on in `tgconfig.json`:

```json
{
  "enumStringInitializers": true
}
```

You can also specify converters that convert C# enum value name to a string initializer (by default string value is exactly the same as C# enum value):

```json
{
  "enumStringInitializersConverters": ["SomeCaseToSomeOtherCase", "..."]
}
```

There is also an attribute you can use on a single C# enum to opt-in/out to the initializers:

```csharp
// opt-in

[TsStringInitializers]
public enum MyEnum {}

// opt-out

[TsStringInitializers(false)]
public enum MyEnum {}
```

On a side note, I also implemented "generation specs" in 2.1.0, which means you can use a configuration file instead of attributes to select C# types to generate:

```csharp
public class MyGenerationSpec : GenerationSpec
{
  public MyGenerationSpec()
  {
    AddClass<MyClass>();
    AddInterface<MyInterface>();
    // ...
  }
}
```

You can read more on them [in the docs](https://typegen.readthedocs.io/en/latest/generationspec.html) - maybe you can find this useful.

Have a happy new year! :)

**Comment by jburzynski** [comment]
Closed due to inactivity.
